tests/integration/valid-projects/rockcraft-complex.yaml

```
@@ -0,0 +1,34 @@
+name: base-2004
+base: ubuntu@20.04
+
+version: "0.1"
+summary: A rock that bundles a Python project.
+description: A rock that bundles a Python project.
+license: GPL-3.0
+platforms:
+  amd64:
+  s390x:
+    build-on: [s390x, arm64]
+    build-for: [s390x]
+  riscv64:
+    build-on: riscv64
+    build-for: riscv64
```

What about testing combinations here?

```suggestion
platforms:
  # Empty platform
  amd64:
  # Build-on, but no build-for, vectorised build-on
  armhf:
    build-on: [arm64, armhf]
  # Build-on but no build-for, scalar
  arm64:
    build-on: amd64
  # Scalar build-for, vector build-on
  i386:
    build-on: [amd64]
    build-for: i386
  s390x:
    build-on: [s390x, arm64]
    build-for: [s390x]
  riscv64:
    build-on: riscv64
    build-for: riscv64
```